src/Build/BackEnd/Components/ProjectCache/ProjectCacheDescriptor.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Graph;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Experimental.ProjectCache
+{
+    public class ProjectCacheDescriptor
+    {
+        public ProjectCacheDescriptor(
+            string pluginPath,
+            IReadOnlyCollection<ProjectGraphEntryPoint>? entryPoints,
+            ProjectGraph? projectGraph,
+            IReadOnlyDictionary<string, string>? pluginSettings = null)
+        {
+            ErrorUtilities.VerifyThrowArgument(
+                (entryPoints == null) ^ (projectGraph == null),
+                "EitherEntryPointsOrTheProjectGraphIsSet");
+
+            PluginPath = pluginPath;
+            EntryPoints = entryPoints;
+            ProjectGraph = projectGraph;
+            PluginSettings = pluginSettings ?? new Dictionary<string, string>();
+        }
+
+        /// <summary>
+        ///     The path to the assembly containing the project cache plugin.
+        /// </summary>
+        public string PluginPath { get; }
+
+        /// <summary>
+        ///     The entry points with which the plugin will be initialized.
+        /// </summary>
+        public IReadOnlyCollection<ProjectGraphEntryPoint>? EntryPoints { get; }
+
+        /// <summary>
+        ///     The graph with which the plugin will be initialized.
+        /// </summary>
+        public ProjectGraph? ProjectGraph { get; }
+
+        public IReadOnlyDictionary<string, string> PluginSettings { get; }
+
+        public override string ToString()
```

Comment:
This seems like a very expensive method, is it going to be called a lot?

Reply:
It's only called once when a BuildManager session loads a plugin. I'll give it a different name to avoid having it called by accident.